script.js

```diff
@@ -1,4 +1,184 @@
+/*Base
+Requirements
+There are 2 players and players take turns.
+When a player clicks Submit, the game rolls 2 dice and shows the dice rolls, for example 3 and 6.
+The player picks the order of the dice they want. For example, if they wanted the number 63, they would specify that the 2nd dice goes first. You can choose how the player specifies dice order.
+After both players have rolled and chosen dice order, the player with the higher combined number wins.*/
+
+// -1: Introduction and game mode selection
+// 0 : Before player 1 roll dice
+// 1 : After player 1 roll dice
+// 2 : Before player 2 roll dice
+// 3 : After player 2 roll dice
+var turnOrOrder = -1;
+
+// 1: Normal mode
+// 2: Reversed Mode
+var gameMode = 0;
+
+// 0 : No action needed
+// 1 : Action needed
+var whichDiceOrder = 0;
+
+// to track overall results
+var pOneStats = 0;
+var pTwoStats = 0;
+var leaderboard = "";
+var myOutputValue;
+
+var playerThrow = [0, 0];
+var pOneResult = 0;
+var pTwoResult = 0;
+
+// Dice roll function to get random dice number
+var diceRollResult = function () {
+  var rollResult = Math.floor(Math.random() * 6) + 1;
+  return rollResult;
+};
+
+// Evaluate results
+var evaluateResults = function (input) {
+  //For both player first roll
+  if ((turnOrOrder == 0 || turnOrOrder == 2) && whichDiceOrder == 0) {
+    whichDiceOrder = 1;
+    turnOrOrder += 1;
+    playerThrow[0] = diceRollResult();
+    playerThrow[1] = diceRollResult();
+    myOutputValue = `You rolled ${playerThrow[0]} and ${playerThrow[1]}.`;
+    console.log(turnOrOrder);
+
+    //If both die result is the same, skip order selection
+    if (playerThrow[0] == playerThrow[1]) {
+      whichDiceOrder = 0;
+      turnOrOrder += 1;
+      //result for P1
+      if (turnOrOrder == 2) {
+        pOneResult = "" + playerThrow[0] + playerThrow[1];
+        //result for P2
+        myOutputValue =
+          myOutputValue +
+          `<br><br>Your hand is ${playerThrow[0]}${playerThrow[1]}.`;
+        return myOutputValue;
+      } else if (turnOrOrder == 4) {
+        pTwoResult = "" + playerThrow[0] + playerThrow[1];
+        myOutputValue =
+          myOutputValue +
+          `<br><br>Your hand is ${playerThrow[0]}${playerThrow[1]}.`;
+        return myOutputValue;
+      }
+    }
```

Comment:
Lines 58-61 and lines 64-67 are repetitive and do the same thing. In programming, we try not to repeat ourselves if possible - the reason for this is 1. for readability purpose, and 2. for code maintainability purpose, imagine you notice a bug here and need to change something, then you need to change it twice for the code sections you repeated. And this gets more complex if there are more instances of unnecessary repetition in your code.

Consider having an if-else that stores the result in either pOneResult or pTwoResult based on the value of turnOrOrder, then doing the other stuff after the if-else.

An example for this section would be:
```
    if (playerThrow[0] == playerThrow[1]) {
      whichDiceOrder = 0;
      turnOrOrder += 1;
      //result for P1
      if (turnOrOrder == 2) {
        pOneResult = "" + playerThrow[0] + playerThrow[1];
      } else if (turnOrOrder == 4) {
        pTwoResult = "" + playerThrow[0] + playerThrow[1];
      }
        myOutputValue =
          myOutputValue +
          `<br><br>Your hand is ${playerThrow[0]}${playerThrow[1]}.`;
        return myOutputValue;
    }
```